base/cvd/cuttlefish/host/commands/cvd/server_command/clear.cpp

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (C) 2024 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "host/commands/cvd/server_command/clear.h"
+
+#include <sys/types.h>
+
+#include <android-base/file.h>
+#include <android-base/parseint.h>
+#include <android-base/scopeguard.h>
+
+#include "common/libs/utils/result.h"
+#include "cuttlefish/host/commands/cvd/cvd_server.pb.h"
+#include "host/commands/cvd/instance_manager.h"
+#include "host/commands/cvd/server_command/server_handler.h"
+#include "host/commands/cvd/server_command/utils.h"
+#include "host/commands/cvd/types.h"
+
+namespace cuttlefish {
+namespace {
+
+constexpr char clearCmd[] = "clear";
+constexpr char kSummaryHelpText[] =
+    "Clears the instance databaase, stopping any running instances first.";
+
+class CvdClearCommandHandler : public CvdServerHandler {
+ public:
+  CvdClearCommandHandler(InstanceManager& instance_manager);
+
+  Result<bool> CanHandle(const RequestWithStdio& request) const override;
+  Result<cvd::Response> Handle(const RequestWithStdio& request) override;
+  cvd_common::Args CmdList() const override;
+  Result<std::string> SummaryHelp() const override;
+  bool ShouldInterceptHelp() const override;
+  Result<std::string> DetailedHelp(std::vector<std::string>&) const override;
+
+ private:
+  InstanceManager& instance_manager_;
+};
+
+CvdClearCommandHandler::CvdClearCommandHandler(
+    InstanceManager& instance_manager)
+    : instance_manager_(instance_manager) {}
+
+Result<bool> CvdClearCommandHandler::CanHandle(
+    const RequestWithStdio& request) const {
+  auto invocation = ParseInvocation(request.Message());
+  return invocation.command == clearCmd;
+}
+
+Result<cvd::Response> CvdClearCommandHandler::Handle(
+    const RequestWithStdio& request) {
+  CF_EXPECT(CanHandle(request));
+
+  cvd::Response response;
+  response.mutable_command_response();
+
+  auto precondition_verified = VerifyPrecondition(request);
```

Comment:
Something for another time, but it would nice if [`VerifyPrecondition`](https://github.com/google/android-cuttlefish/blob/f1e6468df30d3b594776c27a042c452e2d915193/base/cvd/cuttlefish/host/commands/cvd/server_command/utils.cpp#L55) had a more descriptive name.

Reply:
Agreed.
Most likely it should be deleted altogether since all it does is check that ANDROID_HOST_OUT is defined in the environment IIRC